### PR TITLE
Allow users to ignore nicknames

### PR DESCRIPTION
**Pull request by QuiteQuiet**
A bunch of people have requested this, iirc, because they don't want to see their opponent's nicknames but aren't interested in ignoring them entirely.

### CODE REVIEW

**Comment by Zarel** [comment]
I don't think we have space for three checkboxes and a userlist. How does this look?

**Comment by QuiteQuiet** [comment]
Best case | Worst case 


How it looks really depends on how wide the battle log is right now, but that's true for its current state too with just two checkboxes. Ideally they wouldn't be part of the battle log at all, I think.

**Comment by Zarel** [comment]
The top left area is reserved for a userlist...

I think we need to get a better idea of what kind of nicknames are being used that people don't like. I don't think this is necessary.

**Comment by QuiteQuiet** [comment]
My fairly standard laptop resolution (1366x768) already can't fit a userlist in the space that is left over from the Ignore Spectators / Ignore Opponent buttons unless I close any open chat rooms (http://puu.sh/rITcC.jpg) and I'd imagine everyone else with relatively small monitors are already in this situation.

From what I recall there was complaints about things like japaneese nicknames, smileys, and emoticons being used in nicknames. Nothing harmful, just things they don't want to see when they battle.

**Comment by Zarel** [comment]
That seems like it belongs in an Options menu. You'd want it to be turned on before the battle starts, and you'd want it to apply to multiple battles?

**Comment by QuiteQuiet** [comment]
I guess that's a somewhat safe assumption. Though, I don't really know where you'd put another options menu to be honest. I guess as a submenu in the existing options menu would work, but then it'd be nowhere near the other "ignore" options. Alternative would be to put Ignore Opponent/Ignore Spectators in a menu too and add this in there too.

**Comment by Zarel** [comment]
We could replace the checkboxes with an Ignore menu. "In this battle" section, then an "All battles" section.

**Comment by QuiteQuiet** [comment]
Something like this?

File: js/client-battle.js [hunk 1]
```diff
@@ -1110,4 +1098,44 @@
...
+		toggleNicknames: function (side) {
+			for (var i = 0; i < side.active.length; i++) {
+				side.active[i].statbarElem.html(side.getStatbarHTML(side.active[i], true));
+				side.updateStatbar(side.active[i], true, true);
+			}
+			side.updateSidebar();
+		},
+	});
+
+}).call(this, jQuery);
```
Comment (Zarel) [inline js/client-battle.js]: .

Comment (QuiteQuiet) [inline js/client-battle.js]: That's exactly like it is currently too, so what's wrong with it? It just moved down ~30 lines :|

Comment (Zarel) [inline js/client-battle.js]: No, it currently has a trailing linebreak, which you lost.

Trailing linebreaks are bad form; we should probably update eslint to warn about them.

Comment (Zarel) [inline js/client-battle.js]: image
That's what the red symbol at the end is warning about.

Comment (Zarel) [inline js/client-battle.js]: For some reason that symbol isn't showing up in GitHub's "Conversation" tab, but you should be able to tell by the fact that it's green and not white that you changed _something_

File: js/client-battle.js [hunk 2]
```diff
@@ -1105,4 +1098,44 @@
...
+		},
+		toggleIgnoreOpponent: function (e) {
+			this.battle.ignoreOpponent = !!e.currentTarget.checked;
+			this.battle.add('Opponent ' + (this.battle.ignoreOpponent ? '' : 'no longer ') + 'ignored.');
+		},
+		toggleNicknames: function (side) {
+			for (var i = 0; i < side.active.length; i++) {
+				side.active[i].statbarElem.html(side.getStatbarHTML(side.active[i], true));
+				side.updateStatbar(side.active[i], true, true);
+			}
```
Comment (Zarel) [inline js/client-battle.js]: Should you also update the log? That sounds difficult but it might be possible.

Comment (Zarel) [inline js/client-battle.js]: Well, those all _should_ update the log. Especially Ignore Spectators.

Comment (QuiteQuiet) [inline js/client-battle.js]: Do you want it included in this pullreq, or separate?

Comment (Zarel) [inline js/client-battle.js]: Separate is fine.

**Pull request merged**